pageserver/src/tenant/span.rs

```diff
@@ -12,9 +12,6 @@ pub(crate) static TENANT_ID_EXTRACTOR: once_cell::sync::Lazy<MultiNameExtractor<
 #[track_caller]
 pub(crate) fn debug_assert_current_span_has_tenant_id() {
     if let Err(missing) = check_fields_present([&*TENANT_ID_EXTRACTOR]) {
-        panic!(
-            "missing extractors: {:?}",
-            missing.into_iter().map(|e| e.name()).collect::<Vec<_>>()
-        )
+        panic!("missing extractors: {missing:?}")
     }
```

Comment:
```suggestion
    match check_fields_present([&*TENANT_ID_EXTRACTOR]) {
        Ok(Summary::AllPresent) => {},
        Ok(Summary::Unconfigured) && cfg!(test) => {
            // unit tests don't need to have tracing configured
        }
        Ok(Summary::Unconfigured) && !cfg!(test) => {
            panic!("tracing_span_assert requires tracing_error subscriber to be configured")
        }
        Err(missing) => {
            panic!("missing extractors: {missing:?}")
        }
    }
```

Reply:
We don't actually check for the `tracing_error::ErrorLayer` ever. Perhaps it is UNSUPPORTED often but I'll prefer not to add such checks.

Otherwise this looks good ... I'll try to fit it in.

Reply:
I do wish we didn't have to do this, feels a bit like repetition required. I don't think this really changes much, we still do require the matches in tests using the binary. But agreed, me adding an `enum Summary` does sort of tempt this to be extra extra really safe. So it makes me think that's the wrong approach.

`cfg!(test)` is a crate-local, so perhaps having this whole check as a macro in `libs/utils/src/tracing_span_assert.rs` could work.

Or having this `cfg!(test)` as a `bool` parameter to `check_...`, which would also kind of require a macro, but just to expand the `cfg!(test)` elsewhere.

Reply:
Macro in [59f1e79](https://github.com/neondatabase/neon/pull/4682/commits/59f1e790d347bd67b3b1a01850c42d24fb3362d6)